bitseq/sequence.go

```diff
@@ -273,12 +272,9 @@ func (h *Handle) set(ordinal uint32, any bool, release bool) (uint32, error) {
 			continue
 		}
 
-		// Unless unexpected error, save private copy to local copy
+		// Previous atomic push was succesfull. Save private copy to local copy
 		h.Lock()
 		defer h.Unlock()
-		if h.dbIndex != ci {
-			return ret, fmt.Errorf("unexected database index change")
-		}
```

Comment:
@mavenugo @aboch The reason why we make a copy and operate on that is because if two goroutines operate on the same h then what gets stored in the datastore will become inconsistent unless `h.Lock` is held for the entire duration of `set` function. That's why we need to work with a per-thread private copy of `h` and synchronize at the datastore with the help of `PutObjectAtomic`.